### ISSUE TITLE
spring-cloud.version 2020.0.1 seems to break cloud stream

### ISSUE THREAD

**Issue by Mrc0113** [spring-cloud/spring-cloud-stream#2109]
The latest version of spring-cloud seems to break spring cloud stream. Note that I'm using it with the Solace binder, but it doesn't look like a binder specific exception to me. It looks like a conflict between spring cloud stream and spring integration.  

When using `<spring-cloud.version>2020.0.1</spring-cloud.version>` I get the error below on startup of my app. Note that when using the previous version, `2020.0.0`, all works as expected. My assumption is that the issue was introduced with recent changes in this commit: https://github.com/spring-projects/spring-integration/commit/478a79c74bf32cac62d630f1df7c0f520b3e65ca

```
[2m2021-02-05 12:33:48.830[0;39m [33m WARN[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36ms.c.a.AnnotationConfigApplicationContext[0;39m [2m:[0;39m Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Failed to start bean 'inputBindingLifecycle'; nested exception is java.lang.NoSuchMethodError: org.springframework.integration.channel.PublishSubscribeChannel.<init>(Z)V
[2m2021-02-05 12:33:48.832[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.s.c.ThreadPoolTaskScheduler         [0;39m [2m:[0;39m Shutting down ExecutorService 'taskScheduler'
[2m2021-02-05 12:33:48.834[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mc.s.s.c.s.b.SolaceMessageChannelBinder  [0;39m [2m:[0;39m Closing JCSMP session JCSMPSession
[2m2021-02-05 12:33:48.834[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mc.s.j.protocol.impl.TcpClientChannel    [0;39m [2m:[0;39m Channel Closed (smfclient 1)
[2m2021-02-05 12:33:48.838[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.i.monitor.IntegrationMBeanExporter  [0;39m [2m:[0;39m Summary on shutdown: bean 'errorChannel'
[2m2021-02-05 12:33:48.838[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.i.monitor.IntegrationMBeanExporter  [0;39m [2m:[0;39m Summary on shutdown: bean 'log-in-0'
[2m2021-02-05 12:33:48.838[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.i.monitor.IntegrationMBeanExporter  [0;39m [2m:[0;39m Summary on shutdown: nullChannel
[2m2021-02-05 12:33:48.839[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.i.monitor.IntegrationMBeanExporter  [0;39m [2m:[0;39m Summary on shutdown: bean '_org.springframework.integration.errorLogger.handler' for component '_org.springframework.integration.errorLogger'
[2m2021-02-05 12:33:48.844[0;39m [32m INFO[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mConditionEvaluationReportLoggingListener[0;39m [2m:[0;39m 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
[2m2021-02-05 12:33:48.850[0;39m [31mERROR[0;39m [35m84978[0;39m [2m---[0;39m [2m[           main][0;39m [36mo.s.b.d.LoggingFailureAnalysisReporter  [0;39m [2m:[0;39m 

***************************
APPLICATION FAILED TO START
***************************

Description:

An attempt was made to call a method that does not exist. The attempt was made from the following location:

    org.springframework.cloud.stream.binder.BinderErrorChannel.<init>(BinderErrorChannel.java:41)

The following method did not exist:

    org.springframework.integration.channel.PublishSubscribeChannel.<init>(Z)V

The method's class, org.springframework.integration.channel.PublishSubscribeChannel, is available from the following locations:

    jar:file:/Users/XXXXX/.m2/repository/org/springframework/integration/spring-integration-core/5.2.4.RELEASE/spring-integration-core-5.2.4.RELEASE.jar!/org/springframework/integration/channel/PublishSubscribeChannel.class

It was loaded from the following location:

    file:/Users/XXXXX/.m2/repository/org/springframework/integration/spring-integration-core/5.2.4.RELEASE/spring-integration-core-5.2.4.RELEASE.jar


Action:

Correct the classpath of your application so that it contains a single, compatible version of org.springframework.integration.channel.PublishSubscribeChannel
```

**Comment by Mrc0113** [comment]
Also as a heads up, this was brought to my attention by a developer who generated their pom using the latest versions on start.spring.io. 

Tagging @olegz for awareness.

**Comment by olegz** [comment]
@Mrc0113 We've been asked about this internally and externally ever since the release ;)

You need to make sure you have spring-integration 5.4.3 (you probably still have 5.4.2). You should have had it pulled in automatically, with 2020.0.1, but without seeing your POM it's hard to speculate. 
Regardless, this is the issue and once you get the right version of spring-integration all will be well.

And yes it was related to the recent change that went into 3.1.1 - https://github.com/spring-cloud/spring-cloud-stream/issues/2082

**Comment by olegz** [comment]
As far as `start.spring.io.`, i'll test it out as it may still be an issue

**Comment by Mrc0113** [comment]
Thanks @olegz, so instead of starting with their project I just generated a fresh one from start.spring.io and it works for me without having to specify the `spring-integration` version separately. Turns out the culprit was that the developer was using an older version of the `spring-boot-starter-parent` which must have been bringing in an older version of integration.  

Closing this. Thanks again for your help!
